test/toolkits/style_transfer/test_style_transfer.cxx

```diff
@@ -0,0 +1,49 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define BOOST_TEST_MODULE
+
+#include <boost/test/unit_test.hpp>
+#include <core/util/test_macros.hpp>
+#include <toolkits/style_transfer/style_transfer.hpp>
+
+#include <map>
+#include <memory>
+
+#include "utils.hpp"
+
+using namespace turi::style_transfer;
+
+BOOST_AUTO_TEST_CASE(test_train) {
```

Comment:
This PR is already rather big, but we should make some effort to test the functions you're defining. In the future, please consider splitting this work into multiple PRs, where each PR defines one function and its related tests and support code!

Reply:
I didn't expect the train PR to blow up so much. Next time I'll think about what components go into files a bit more. I did it on a macro level with model_spec, compute_context and data loaders. I'll figure out how to do it on a micro scale per file. I added a couple of more tests, nothing crazy just checking no see-faults. I'll add more for the predict PR.